### DISCUSSION TITLE
How do i use the Get-Command command to grab a certain type of commands

### DISCUSSION THREAD

**Discussion by hellaAnonymous** [PowerShell/PowerShell#15348] [General]
I know that using the Get-Command cmdlet on its own, will grab all of the commands (cmdlets, and functions), and that using for example "Get-Command "Clear"" it displays:

 Alias           

clear -> Clear-Host" 

which then in order to see the Version, of that object since it is an alias you have to type "Get-Command "Clear-Host"" which will not display the Version or source since it is a function not a cmdlet (in this example). 

But what I want to be able to do is use "Get-Command" + for example "CmdType" or "CommandType" = "cmdlet, or function" to just grab the commands of that specified type.

**Comment by vexx32** [comment]
```
Get-Command -CommandType Cmdlet
```

(As an example).

I'd recommend reading the help content as well; `Get-Help Get-Command -Full`. 🙂

**Reply by hellaAnonymous** [reply]
ok
thanks

**Comment by AltitudeApps** [comment]
Even if `Get-Command` did not support the `-CommandType` option, you could use `Where-Object` to filter the results yourself:
```pwsh
get-command | Where-Object { $_.CommandType -eq 'Alias' }
```
